packages/image_picker/image_picker_platform_interface/lib/src/method_channel/method_channel_image_picker.dart

```diff
@@ -280,13 +308,17 @@ class MethodChannelImagePicker extends ImagePickerPlatform {
     assert(result.containsKey('path') != result.containsKey('errorCode'));
 
     final String? type = result['type'] as String?;
-    assert(type == kTypeImage || type == kTypeVideo);
+    assert(
+      type == kTypeImage || type == kTypeVideo || type == kTypeMedia,
+    );
 
     RetrieveType? retrieveType;
     if (type == kTypeImage) {
       retrieveType = RetrieveType.image;
     } else if (type == kTypeVideo) {
       retrieveType = RetrieveType.video;
+    } else if (type == kTypeMedia) {
+      retrieveType = RetrieveType.media;
     }
```

Comment:
It seems like this would be better as a switch rather than an assert and if/else if chain, so that the compile alerts us to missing cases.